### ISSUE TITLE
[1.12.2 - Redstone Arsenal 2.3.10.4] Flux-Infused Shield spider/mite/silverfish weakness

### ISSUE THREAD

**Issue by VallenFrostweaver** [CoFH/Feedback#861]
When defending against spiders/endermites/silverfish (empowered or not) it does no damage to them. Works fine against other entities.
EDIT: Slimes too.

**Comment by KingLemming** [comment]
So, this is upstream in nature. I don't know exactly where, but it's either that those mobs don't fire a LivingAttackEvent or when they do, they report a null source, which means they are untargetable in the resolution of said event.

**Comment by VallenFrostweaver** [comment]
They suffer from thorns enchants though but then again they are also dealing damage so perhaps that is different.

**Comment by KingLemming** [comment]
Thorns is handled through a different route, but I did some testing and noticed that slimes DO in fact trigger the effect, but only slimes above a certain size. So I know what might be causing it.

**Comment by KingLemming** [comment]
Okay, so I've handled this on my end. The issue was basically that I had to work around vanilla logic when I made shields as Forge did not properly support them. In doing so, I copied vanilla's logic in my EventHandler straight up, which meant that shields aren't "used" if an attack is less than 3 damage.

So now there's an "OnHit" for this case.

It works with almost everything, but I can't get tiny slimes to trigger it. For some reason or other, they fire the attack event and the attacked entity is null, not a player. I'm pretty sure if the root cause of that is fixed, then it will work as well.
